data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/DataPrepperServer.java

```diff
@@ -48,4 +65,42 @@ public void stop() {
         server.stop(0);
         LOG.info("Data Prepper server stopped");
     }
+
+    private HttpServer createHttpServer(final int port) throws IOException {
+        return HttpServer.create(new InetSocketAddress(port), 0);
+    }
+
+    private HttpServer createHttpsServer(final int port,
+                                         final String keystoreFilePath,
+                                         final String passphrase) throws IOException {
+        final SSLContext sslContext;
+
+        try {
+            char[] passphraseArray = passphrase.toCharArray();
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(new FileInputStream(keystoreFilePath), passphraseArray);
```

Comment:
another nit: can we consider using `Files.newInputStream` over `new FileInputStream`